src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1661,6 +1662,51 @@ ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
     register_matcher(std::make_shared<opp::Matcher>(transpose_out, "ConvToMatmul"), std::move(callback));
 }
 
+// Removing uneffective zeropoint layer, specific structure fixed for POC only purpose,
+// further all unefective subtract can be optimized-out
+// anyinput[2D], "uneffective_zp_subgraph"[1D], subtract(boardcast), any_input -> multiply, transpose -> convolution
+// anyinput[2D], any_input -> multiply, transpose -> convolution
+
+UneffectiveZP::UneffectiveZP() {
+    // Constant / param
+    auto zero_point_tensor = opp::any_input();
+    auto sub_bias_convert = opp::optional<ov::op::v0::Convert>({zero_point_tensor->outputs()});
+    auto sub_bias_reshape = opp::optional<ov::op::v1::Reshape>({sub_bias_convert, opp::any_input()});
+    auto sub_weights = opp::any_input();
+    auto subtract = opp::wrap_type<ov::op::v1::Subtract>({sub_weights, sub_bias_reshape});
+    auto multiply_bias = opp::any_input();
+    auto multiply = opp::wrap_type<ov::op::v1::Multiply>({subtract, multiply_bias});
+    auto conv_any_bias = opp::any_input();
+    auto conv = opp::wrap_type<ov::op::v1::Convolution>({conv_any_bias, multiply});
+
+    auto callback = [=](ov::pass::pattern::Matcher& m) -> bool {
+        LOG_DEBUG("UneffectiveZP subgraph candidate detected");
+        auto& node_to_output = m.get_pattern_value_map();
+        auto zero_point_tensor_node = node_to_output.at(zero_point_tensor).get_node_shared_ptr();
+
+        if (op::util::is_constant_and_all_values_equal_int(zero_point_tensor_node, 0)) {
+            LOG_DEBUG("UneffectiveZP subgraph detected with NOP zp tensor: "
+                << zero_point_tensor_node->get_friendly_name() << " shape=" <<  zero_point_tensor_node->get_shape());
+            // removing whole zubgraph with subtract + zeropoints
+            auto multiply_node = node_to_output.at(subtract).get_node_shared_ptr();
+            auto sub_weights_node = node_to_output.at(sub_weights).get_node_shared_ptr();
+
+            //TODO: verify input 0 is subtract
+            multiply_node->input(0).replace_source_output(sub_weights_node->output(0));
```

Comment:
`->input(0)` ?

Reply:
->input(0) seems ok - i think i've tried both cases, sub_weights_node and sub_weights_node->output(0), as well. Will check one more time.